src/kio/serial/_parse.py

```diff
@@ -126,8 +134,24 @@ def get_field_reader(
 E = TypeVar("E", bound=Entity)
 
 
-@cache
+@overload
 def entity_reader(entity_type: type[E]) -> readers.Reader[E]:
+    ...
+
+
+@overload
+def entity_reader(
+    entity_type: type[E],
+    nullable: Literal[True],
+) -> readers.Reader[E | None]:
+    ...
```

Comment:
What's the need for these overloads?

Reply:
The overload reflects that if you call with `nullable=True` return type has `E | None` otherwise just `E`. 

Am I correct in guessing that the question is referring to the fact that one of the call-sites is messy, and doesn't care about the difference?

These functions are also exposed as public APIs, and for those uses it's important that applications don't need to eliminate `None` in the `nullable=False` cases.

Reply:
It _might_ be possible to address the messy call-site. The culprit is making mypy narrow the T typevar using the call to `classify_field()`. I think this might be possible if that function was redesigned to return some generic object rather than a tuple ... 🤔